### ISSUE TITLE
Concider doing an ARM64 build as well ..

### ISSUE THREAD

**Issue by bmentink** [helloSystem/hello#190]
More and more computers are going to arm64, including Macos M1, Windows, single board computers like RaspberryPi 4 etc

The linux arm64 kernel already has support for Macos M1, so it won't be too long before Linux is able to run on that hardware. I would love to see, in the long term, hellosystem running on the M1, but at the moment I would be happy to have a play with hello on the Raspberry Pi 4, oron my Rock64 board.

I have seen the build instructions for the Pi 4, so I could manually do it, but it means a lot of work to upgrade .. how hard is it to add build for arm64 to the build script?

**Comment by probonopd** [comment]
Once https://github.com/cirruslabs/ supports it. Maybe @fkorotkov knows?

**Comment by fkorotkov** [comment]
Expect some news in the next couple of weeks 😉

**Comment by bmentink** [comment]
Cool 😎😎

**Comment by bmentink** [comment]
Hi Guys,

I have just received a Khadas  Edge-V board  [https://docs.khadas.com/edge/#Edge-V](url)
I have installed a FreeBSD image that has support for sound over HDMI, custom UBoot etc and a few other tweaks. Tis is provided my a forum member over there **Sleepwalker**

I have attempted to do the manual raspberrypi style installation of hellosystem as per [https://hellosystem.github.io/docs/developer/rpi.html](url) but am having some issues.

When I "startx" it complains of missing /System/Kwin app and also /System/Filer.AppDir and /System/Menu.AppDir.
I did not see any reference to these in the instructions.

I had a look at the "start-hello" script and found those items and installed them in /System, but I suspect that they contain x86 binaries ...

Can someone help with getting these items for ARM64 ..

Cheers

**Comment by probonopd** [comment]
You need to write your own minimal `/usr/local/bin/start-hello`.

**Comment by bmentink** [comment]
So, I am a bit confused. The instructions showing how to bring hello up on a raspberrypi show it actually running. So I am assuming someone has an ARM64 version of Filer.AppDir,  Menu.AppDir and KWin.app somewhere. Knowing where these are would be a start. Also, there needs to be an arm version of gmenudbusmenuproxy it seems ..

Cheers

**Comment by linuxkettle** [comment]
They're in /System. You have to compile them yourself, see the CI/CD scripts.

**Comment by linuxkettle** [comment]
We use Cirrus, don't we?

**Comment by bmentink** [comment]
@linuxkettle 
I am not a developer, and therefor have no idea what a CI/CD script is. I am just helping by testing images.

I will wait for @fkorotkov to make his announcement ... hoping that he has RK3399 support ;)

**Comment by linuxkettle** [comment]
Each of these repositories (except Utilities) has been compiled for x86, not arm. So somebody needs to compile them for arm manually, which is no easy task.

We could use multi arch app bundles...

**Comment by bmentink** [comment]
@linuxkettle 
Agree, multi arch app is a good solution. Especially as ARM64 and Risc-V will be the mainstream CPU's in the not so distant future, x86 is on the way out, Intel can never recover ... my 2c. :)

**Comment by linuxkettle** [comment]
I have made multi-arch AppImages before, you could apply the same logic to app bundles...

**Comment by robertoandrade** [comment]
any news on this? trying to run this in a VM on a M1 mac, sure could run the x86 version instead but wanted to get the performance boost.

**Comment by probonopd** [comment]
Depends on:
* https://github.com/cirruslabs/cirrus-ci-docs/issues/906
